New Checks: semantics validation on attribute reference (#163)
bflad/tfproviderlint: passes/S038/README.md

# S038

> The S038 analyzer reports cases of Schemas which include `ConflictsWith` and have invalid schema attribute references.

Reviewer: ```suggestion
The S038 analyzer reports cases of Schemas which include `AtLeastOneOf` and have invalid schema attribute references.
```

Author: Sorry for the mistake 😅

> NOTE: This pass only works with Terraform resources that are fully defined in a single function.

Reviewer: Terraform resources that meet the criteria for this check can come from a variety of source code, e.g. variables, inline declarations in `schema.Provider.ResourcesMap`, etc.

```suggestion
NOTE: This pass only works with `schema.Resource` that are wholly declared.
```

Author: You're right

## Flagged Code

```go

> // Attribute reference in multi nested block is not supported

Reviewer: ```suggestion
// Attribute reference in configuration block with potentially more than one element is not supported
```
